packages/enhanced/src/lib/container/HoistContainerReferencesPlugin.ts

```diff
@@ -3,78 +3,77 @@ import type {
   Compilation,
   Chunk,
   WebpackPluginInstance,
+  Module,
 } from 'webpack';
-import ContainerEntryModule from './ContainerEntryModule';
 
-/**
- * This class is used to hoist container references in the code.
- * @constructor
- */
 export class HoistContainerReferences implements WebpackPluginInstance {
-  private containerName?: string | undefined;
+  private readonly containerName: string;
 
-  constructor(name?: string | undefined) {
-    this.containerName = name;
+  constructor(name?: string) {
+    this.containerName = name || 'no known chunk name';
   }
 
   apply(compiler: Compiler): void {
     compiler.hooks.thisCompilation.tap(
       'HoistContainerReferences',
       (compilation: Compilation) => {
-        compilation.hooks.afterOptimizeChunks.tap(
-          'HoistContainerReferences',
-          (chunks: Iterable<Chunk>) => {
-            for (const chunk of chunks) {
-              if (this.chunkContainsContainerEntryModule(chunk, compilation)) {
-                this.hoistModulesInChunk(chunk, compilation);
-              }
-            }
-          },
-        );
+        compilation.hooks.afterChunks.tap('HoistContainerReferences', () => {
```

Comment:
execute hoist before module optimiization takes place, this allows us to hoist module before chunk generation and reduce update calls to the chunk graph which can cause more hmr updates